crates/sncast/src/state/state_file.rs

```diff
@@ -1,13 +1,82 @@
 #![allow(dead_code)]
 use crate::helpers::constants::STATE_FILE_VERSION;
+use crate::response::errors::StarknetCommandError;
 use crate::response::structs::{DeclareResponse, DeployResponse, InvokeResponse};
 use crate::state::hashing::generate_id;
+use crate::WaitForTransactionError;
 use anyhow::{anyhow, Context, Result};
+use cairo_felt::Felt252;
 use camino::Utf8PathBuf;
+use conversions::felt252::SerializeAsFelt252Vec;
 use serde::{Deserialize, Serialize};
 use serde_json::Value;
 use std::collections::HashMap;
 use std::fs;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+struct InnerStateManager {
+    state_file: Utf8PathBuf,
+    executed_transactions_prev_run: ScriptTransactionEntries,
+    executed_transactions_current_run: ScriptTransactionEntries,
+}
+
+#[derive(Default)]
+pub struct StateManager {
+    inner: Option<InnerStateManager>,
```

Comment:
Why this is `Option`?

Reply:
I would lean towards having `StateManager` that is `None` in such cases, rather than `StateManager` devoid of actual logic. The reasoning behind this design isn't immediately apparent when looking at the code, although I get that it might have been simpler to implement it this way.

Reply:
this is why we derive `Default` in here - so it is None i such cases :+1: